### DISCUSSION TITLE
WebAPI Optimization

### DISCUSSION THREAD

**Discussion by HankChill** [RetroAchievements/RAWeb#732] [Ideas]
Currently the Web APIs for retrieving extended game information can only be called with a single game ID. This makes for scraping game data a lengthy task, with more calls than necessary, as in order to scrape game data with achievements the following tasks have to be done:

1) Call the API to retrieve all games for a platform. There is no parameter that offers a filter for games that have achievements, so the list of games to get info for is much larger than it needs to be.

2) For each game in the game list retrieved from step 1, the API for getting extended game info needs to be called for every game individually. Again, no parameter that provides a CSV value for retrieving all game details at once. This results in many more API calls than necessary. If a game doesn't contain achievements, then it's a wasted API call and extra traffic that isn't necessary. This is mostly the case, especially with platforms that have 3000+ titles like the NES, as only 452 titles contain achievements. that's almost 3,500 wasted API transactions.

**Comment by Jamiras** [comment]
`API_GetGameList` can be called with `f=1` to only return games with achievements. This was added in 2020 by https://github.com/RetroAchievements/RAWeb/commit/6e88f1812d62061e717309c013b9a4b29e41db17

Note that this will only return the list of games with achievements. If you want the achievements themselves, you still have to make additional calls for each game.
